digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala

```diff
@@ -438,6 +438,94 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: KgvRoadLink,
     }
   }
 
+  private val getDataForRoadAddressChangesBrowser: SwaggerSupportSyntax.OperationBuilder = (
+    apiOperation[Map[String,Any]]("getDataForRoadAddressChangesBrowser").parameters(
+      queryParam[String]("startDate").description("Start date (yyyy-MM-dd)"),
+      queryParam[String]("endDate").description("End date (yyyy-MM-dd)"),
+      queryParam[String]("dateTarget").description("What start and end dates are used for"),
+      queryParam[Long]("ely").description("Ely number of a road address").optional,
+      queryParam[Long]("roadNumber").description("Road Number of a road address").optional,
+      queryParam[Long]("minRoadPartNumber").description("Min Road Part Number of a road address").optional,
+      queryParam[Long]("maxRoadPartNumber").description("Max Road Part Number of a road address").optional
+    )
+      tags "ViiteAPI - Road Address Changes Browser"
+      summary "Returns change info for road address changes browser based on the search criteria"
+    )
+
+  get("/roadaddresschangesbrowser", operation(getDataForRoadAddressChangesBrowser)) {
+    time(logger, s"GET request for /roadaddresschangesbrowser") {
+      def validateInputs(startDate: Option[String], endDate: Option[String], dateTarget: Option[String], ely: Option[Long], roadNumber: Option[Long], minRoadPartNumber: Option[Long], maxRoadPartNumber: Option[Long]): Boolean = {
+        def parseDate(dateString: Option[String]): Option[DateTime] = {
+          val formatter = DateTimeFormat.forPattern("yyyy-MM-dd")
+          try {
+            if (dateString.isDefined) {
+              Some(formatter.parseDateTime(dateString.get))
+            } else
+              None
+          } catch {
+            case _: IllegalArgumentException => None
+          }
+        }
+
+        def roadPartInputsValid(minRoadPartNumber: Option[Long], maxRoadPartNumber: Option[Long]) = {
```

Comment:
eikö teilllä ole sellaista roadpart class tai object jossa olisi validointi metodi kun objectin luo?

Reply:
tää niin tuntuu sellaiselta koodilta joka kuuluisi olla jonkun domain object sisällä ja tarkistamassa että tietomalli on validi

Reply:
Tälläisillä domain object validointi koodeilla on ikävä tapa duplikoitua ympäriinsä ja pienillä muutoksilla sinne ja tänne.

Reply:
Keskusteltiin perusluokkien, kuten tieosoite (roadAddress), implementoinnista. 
Eka steppi on jo tuolla... RoadAddressBrowserissakohan se oli. 
Mutta todettiin, että täytyy tehdä erillisellä tiketillä eteenpäin; jo toinen steppi ois vieny saman tien syvälle kaninkoloon.